src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -2885,7 +2885,10 @@ internal override AnalyzerDriver AnalyzerForLanguage(ImmutableArray<DiagnosticAn
 
         internal void SymbolDeclaredEvent(Symbol symbol)
         {
-            if (EventQueue != null) EventQueue.Enqueue(new SymbolDeclaredCompilationEvent(this, symbol));
+            if (EventQueue?.IsCompleted == false)
+            {
+                EventQueue.Enqueue(new SymbolDeclaredCompilationEvent(this, symbol));
```

Comment:
Perhaps we shouldn't raise events for unreachable symbols even when queue is still "open".